hyperspy/_components/doniach.py

```diff
@@ -150,19 +150,25 @@ def estimate_parameters(self, signal, x1, x2, only_current=False):
             self.centre.value = centre
             self.sigma.value = sigma
             self.A.value = height * 1.3
-            if is_binned(signal) is True:
+            if is_binned(signal):
             # in v2 replace by
             #if axis.is_binned:
-                self.A.value /= axis.scale
+                if axis.is_uniform:
+                    self.A.value /= axis.scale
+                else:
+                    self.A.value /= np.gradient(axis.axis)[axis.value2index(centre)]
```

Comment:
This could simplify to:
```suggestion
                # use gradient to support non uniform axis
                self.A.value /= np.gradient(axis.axis)[axis.value2index(centre)]
```

Reply:
I can do so for all the occurences, just thought it might be computationally more easy to stay with the scale for the uniform axes.

Reply:
Fair point, it is in the order of 10s of us, which is still very small and `estimate_parameters` is not expected to be called many many times. Maybe an alternative to avoid repetition is to define a scaling factor before the `if only_current is True:`:
```python
factor = 1
if is_binned(signal):
   if axis.is_uniform:
      factor = axis.scale
  else:
      factor = np.gradient(axis.axis)[axis.value2index(centre)]

if only_current is True:
   ...
   self.A.value /= factor
```

Similarly as you did for the test.